apps/api/src/controllers/conversations.rs

```diff
@@ -5,19 +5,61 @@ use actix_web::{web, HttpResponse, Responder};
 use crate::auth_extractor::AuthenticatedUser;
 use crate::models::{CreateConversation, NewMessage};
 use crate::services::conversation::ConversationService;
+use crate::services::ollama::OllamaService;
 use serde::Deserialize;
+use tracing::error;
+use std::collections::HashSet;
 
 /// POST `/api/conversations` - create a new conversation for the current user.
 pub async fn create_conversation(
     service: web::Data<ConversationService>,
+    ollama: web::Data<OllamaService>,
     user: AuthenticatedUser,
     payload: web::Json<CreateConversation>,
 ) -> impl Responder {
+    // Validate input manually
+    let req = payload.into_inner();
+    if let Some(ref model) = req.model {
+        if model.len() > 50 {
+            return HttpResponse::BadRequest().body("Model name must be less than 50 characters");
+        }
+        // Validate that the model exists
+        match ollama.list_local_models().await {
+            Ok(models_json) => {
+                if let Some(models) = models_json.get("models") {
+                    if let Some(models_array) = models.as_array() {
+                        let available_models: HashSet<String> = models_array
+                            .iter()
+                            .filter_map(|m| m.get("name").and_then(|n| n.as_str()))
+                            .map(|s| s.to_string())
+                            .collect();
+                        
+                        if !available_models.contains(model) {
+                            return HttpResponse::BadRequest().body("Invalid model name");
+                        }
+                    }
+                }
+            }
+            Err(_) => {
+                // If we can't verify the model, reject the request
+                return HttpResponse::BadRequest().body("Unable to verify model availability");
+            }
+        }
+    }
```

Comment:
Il existe peut-être une meilleur façon de faire cela ? Visiblement le coût de construction de la hashmap est supérieur à une recherche par itération.